rotkehlchen/tests/unit/decoders/test_velodrome.py

```diff
@@ -0,0 +1,832 @@
+            address=ROUTER_V1,
+            notes=F'Receive 0.181508840163183003 ETH as the result of a swap in {CPT_VELODROME}',
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('optimism_accounts', [['0x78C13393Aee675DD7ED07ce992210750D1F5dB88']])
+def test_swap_tokens(optimism_accounts, optimism_transaction_decoder):
+    """Check that swapping tokens in velodrome is properly decoded."""
+    evmhash = deserialize_evm_tx_hash('0x0cca192bce4a72059c255e5000a0ba1e716f1059d2a388167165e41d9269aa24')  # noqa: E501
+    user_address = optimism_accounts[0]
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=optimism_transaction_decoder.evm_inquirer,
+        database=optimism_transaction_decoder.database,
+        tx_hash=evmhash,
+    )
+    timestamp = TimestampMS(1694638139000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(FVal('0.000051606343716023')),
+            location_label=user_address,
+            counterparty=CPT_GAS,
+            notes='Burned 0.000051606343716023 ETH for gas',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=96,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=A_OP,
+            balance=Balance(FVal('0')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=ROUTER_V2,
+            notes=F'Revoke OP spending approval of {user_address} by {ROUTER_V2}',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=97,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.SPEND,
+            asset=A_OP,
+            balance=Balance(FVal('1.5')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=WETH_OP_POOL_ADDRESS,
+            notes=f'Swap 1.5 OP in {CPT_VELODROME}',
+        ), EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=109,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.TRADE,
+            event_subtype=HistoryEventSubType.RECEIVE,
+            asset=Asset(VELO_V2_TOKEN),
+            balance=Balance(FVal('50.633275729041219862')),
+            location_label=user_address,
+            counterparty=CPT_VELODROME,
+            address=string_to_evm_address('0x0af32614EEa68B8D2232B9592FbdB6512ab6DA73'),  # fbomb_velo_pool_address  # noqa: E501
```

Comment:
What's fbomb? Just the token name?

Reply:
yes just a token name. I removed that comment since it is not something we should care I guess.